### PR TITLE
feat(tool,nextcloud)!: Generate rich object parameters

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/nextcloud/neon/issues/1847

Not really useful at the moment because of the lack of discriminator support (https://github.com/nextcloud/neon/issues/1231) and a way to use exhaustive switch statements.

### CODE REVIEW

**Comment by provokateurin** [comment]
I change the patch to only be a simple object instead of the discriminator so in theory it is usable now (and also everything works). I would like to first upstream the missing parameter type and missing fields. Additionally we have to wait until the next Talk releases which will include https://github.com/nextcloud/spreed/pull/12021 as the some parameters were incorrectly not typed as string before.

**Comment by provokateurin** [comment]
Ok apparently the fix for Talk was only backport to 29. To keep supporting the broken values we'll have to change the type to `oneOf<int|string>` for now :/ Once 28 support is dropped we can remove the hack.

**Comment by provokateurin** [comment]
PR for the fields that are missing in the current definitions is sent upstream: https://github.com/nextcloud/server/pull/45424

**Comment by provokateurin** [comment]
Let's just get this merged and using the discriminator can be done later once dynamite properly supports it again.

File: packages/nextcloud/lib/src/api/spreed.openapi.json [hunk 1]
```diff
@@ -900,97 +900,139 @@
...
-                                "type": "string"
-                            },
-                            {
-                                "type": "integer"
-                            }
-                        ]
+                        "type": "string",
+                        "description": "The file size in bytes",
+                        "example": "3145728"
                     },
```
Comment (Leptopoda) [inline packages/nextcloud/lib/src/api/spreed.openapi.json:971]: Why can't it be an integer anymore?
Is this an error or was it just false before?

**Review by Leptopoda — COMMENTED** [review]
LGTM. Please verify the comment.

File: packages/nextcloud/lib/src/api/spreed.openapi.json (hunk 1 above)
Comment (provokateurin) [reply packages/nextcloud/lib/src/api/spreed.openapi.json:971]: Ah yeah I forgot to patch that back in

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**